spec/support/helpers.rb

```diff
@@ -217,10 +217,21 @@ def gem_command(command, args = "", options = {})
     end
     bang :gem_command
 
+    def test_signals
+      open3_reserved_signals = %w[CHLD CLD PIPE]
+      reserved_signals = %w[SEGV BUS ILL FPE VTALRM KILL STOP EXIT]
+      bundler_signals = %w[INT]
+
+      Signal.list.keys - (bundler_signals + reserved_signals + open3_reserved_signals)
+    end
+
     def sys_exec(cmd)
       command_execution = CommandExecution.new(cmd.to_s, Dir.pwd)
 
       Open3.popen3(cmd.to_s) do |stdin, stdout, stderr, wait_thr|
+        test_signals.each do |n|
```

Comment:
this probably shouldn't be executed on every call to `sys_exec`?

Reply:
I didn't see any concerns, but fair, moved to `bundler`, as its more clear now.

Reply:
No, I mean, shouldn't this just be done for the single test case?

Reply:
um! yeah we could, though I figured this would help have a consistent implementation of bundle exec in specs, when it comes to Signals. Do you anticipate concerns?

Reply:
I updated so that signal trapping is now specific to the spec itself.